src/runtime/eventobject.cs

```diff
@@ -72,17 +72,22 @@ internal bool AddEventHandler(IntPtr target, IntPtr handler)
         /// </summary>
         internal bool RemoveEventHandler(IntPtr target, IntPtr handler)
         {
+            if (reg == null)
+            {
+                Exceptions.SetError(Exceptions.ValueError, "unknown event handler");
+                return false;
+            }
+
             object obj = null;
             if (target != IntPtr.Zero)
             {
                 var co = (CLRObject)GetManagedObject(target);
                 obj = co.inst;
             }
 
-            IntPtr hash = Runtime.PyObject_Hash(handler);
-            if (Exceptions.ErrorOccurred() || reg == null)
+            nint hash = Runtime.PyObject_Hash(handler);
+            if (hash == -1 || Exceptions.ErrorOccurred())
```

Comment:
It must be `&&`, not `||`. `PyObject_Hash` on error returns `-1` (but also when no error and hash is -1), so if `hash` is not `-1`, there's no error.